lib/content-services/src/lib/common/services/rendition.services.spec.ts

```diff
@@ -0,0 +1,105 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { AlfrescoApiService, LogService, TranslationService, ViewUtilService } from '@alfresco/adf-core';
+import { Rendition, RenditionEntry, RenditionPaging, RenditionsApi } from '@alfresco/js-api';
+import { RenditionService } from '@alfresco/adf-content-services';
+
+const getRenditionEntry = (status: Rendition.StatusEnum): RenditionEntry => {
+    return {
+        entry: {
+            id: 'pdf',
+            status: status,
+            content: { mimeType: 'application/pdf', mimeTypeName: 'application/pdf', sizeInBytes: 10 }
+        }
+    };
+};
+const getRenditionPaging = (status: Rendition.StatusEnum): RenditionPaging =>  {
+    return {
+        list: {
+            entries: [getRenditionEntry(status)]
+        }
+    };
+};
+describe('RenditionService', () => {
+    let renditionService: RenditionService;
+    let renditionsApi: any;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            providers: [
+                RenditionService,
+                { provide: AlfrescoApiService, useValue: {} },
+                { provide: LogService, useValue: { error: jasmine.createSpy('error') } },
+                { provide: TranslationService, useValue: {} },
+                { provide: ViewUtilService, useValue: {} },
+                { provide: RenditionsApi, useValue: {
+                        listRenditions: jasmine.createSpy('listRenditions'),
+                        getRendition: jasmine.createSpy('getRendition'),
+                        createRendition: jasmine.createSpy('createRendition')
+                    } }
+            ]
+        });
+        renditionService = TestBed.inject(RenditionService);
+        renditionsApi = TestBed.inject(RenditionsApi);
+        // eslint-disable-next-line no-underscore-dangle
+        renditionService._renditionsApi = renditionsApi;
```

Comment:
this is a hack, the `_renditionsApi` should be private, you can use `spyOnProperty(renditionService, 'renditionsApi').and.returnValue(renditionsApi)`

Reply:
otherwise, you are already using provider with a custom Renditions api, so it should automatically be available